src/Umbraco.Core/Cache/ObjectCacheAppCache.cs

```diff
@@ -1,31 +1,53 @@
-using System.Runtime.Caching;
 using System.Text.RegularExpressions;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
 using Umbraco.Extensions;
 
 namespace Umbraco.Cms.Core.Cache;
 
 /// <summary>
-///     Implements <see cref="IAppPolicyCache" /> on top of a <see cref="ObjectCache" />.
+///     Implements <see cref="IAppPolicyCache" /> on top of a <see cref="MemoryCache" />.
 /// </summary>
 public class ObjectCacheAppCache : IAppPolicyCache, IDisposable
 {
+    private readonly IOptions<MemoryCacheOptions> _options;
+    private readonly IHostEnvironment? _hostEnvironment;
+    private readonly ISet<string> _keys = new HashSet<string>();
     private readonly ReaderWriterLockSlim _locker = new(LockRecursionPolicy.SupportsRecursion);
     private bool _disposedValue;
 
     /// <summary>
-    ///     Initializes a new instance of the <see cref="ObjectCacheAppCache" />.
+    /// Gets the internal memory cache, for tests only!
     /// </summary>
-    public ObjectCacheAppCache() =>
+    /// <value>
+    /// The memory cache.
+    /// </value>
+    internal MemoryCache MemoryCache { get; private set; }
 
-        // the MemoryCache is created with name "in-memory". That name is
-        // used to retrieve configuration options. It does not identify the memory cache, i.e.
-        // each instance of this class has its own, independent, memory cache.
-        MemoryCache = new MemoryCache("in-memory");
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ObjectCacheAppCache" />.
+    /// </summary>
+    public ObjectCacheAppCache()
+        : this(Options.Create(new MemoryCacheOptions()), NullLoggerFactory.Instance, null)
+    { }
 
     /// <summary>
-    ///     Gets the internal memory cache, for tests only!
+    /// Initializes a new instance of the <see cref="ObjectCacheAppCache" /> class.
     /// </summary>
-    public ObjectCache MemoryCache { get; private set; }
+    /// <param name="options">The options.</param>
+    /// <param name="loggerFactory">The logger factory.</param>
+    /// <param name="hostEnvironment">The host environment.</param>
+    public ObjectCacheAppCache(IOptions<MemoryCacheOptions> options, ILoggerFactory loggerFactory, IHostEnvironment? hostEnvironment)
```

Comment:
Although this potentially allows providing custom options, the `AppCaches` instance is manually created when still configuring services (and registered in DI as a singleton instance).

This also means that the `IHostEnvironment` is always null and therefore the file based expiration doesn't actually work... There's also no tests for this, so maybe it's better to simply remove it altogether (potentially also the `dependentFiles` parameter, although that would introduce more breaking changes).

Reply:
Since this PR already contains a breaking change, we don't use the `dependentFiles` anywhere in our code and it doesn't work because the `IHostEnvironment` parameter is always null and on top of that there are no tests, removing it seems the most logical thing to do...

If a user needs this feature, they can just as easily inject `IMemoryCache` instead of `AppCaches` (and also avoid an additional level of abstraction and a direct dependency on the CMS).